src/aleph_p2p_client/client.py

```diff
@@ -70,7 +70,8 @@ async def receive_messages(
         async with sub_queue.iterator() as queue_iter:
             async for message in queue_iter:
                 yield message
-                await message.ack()
+                if not message.processed:
```

Comment:
This is not blocking and can be done later but I would add logging here for easier debugging.

I would also add a comment if possible? It doesn't look super logical at first glance to ack a message only if it's not processed '-'